app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java

```diff
@@ -341,19 +371,42 @@ public void onScanResult(int callbackType, ScanResult result) {
                         caller.register(callBack, result.getDevice(), device, txt);
 
                         Log.d("DEBUG:", "Address:" + result.getDevice().getAddress());
-
+                        deviceAddress = result.getDevice().getAddress();
+                        scanLeDevice(false, null,null);
                     }
                 }
             }
         };
 
+    private ScanCallback mSummaryScanCallback = new ScanCallback() {
+
+        @Override
+        public void onScanResult(int callbackType, ScanResult result) {
+            super.onScanResult(callbackType, result);
+            ScanRecord record = result.getScanRecord();
+            String name;
+            if(record == null) return;
+            else{
+                name = record.getDeviceName();
+            }
+
+            if (name == null) {
+                Log.i("onScanResult", "scanned device name null");
+            } else {
+                if(isEspertoWatch(name)){
+                    Log.i("onScanResult", "Esperto watch found");
+                    scanFromSummary(false);
+                }
+            }
+        }
+    };
+
     //function to filter out force sensors
     public boolean isEspertoWatch(String name)
     {
         //TODO::update with new BLE chip
-        String watch_name = getString(R.string.esperto_device_name);
-        if(name.equals(watch_name)) return true;
-        else return false;
+        String watchName = getString(R.string.esperto_device_name);
+        return name.equals(watchName);
```

Comment:
Do we really need this to be its own function, we could just do if(name.equals(getString(R.string.esperto_device_name)))

Reply:
This is pretty minor though so it's fine to just not change if you don't feel like it, also I guess it sorta improves code readability

Reply:
Probably doesn't need to be its own method.

Reply:
Looks like its being used in tests. Keeping as is.